src/actor/actor.hpp

```diff
@@ -39,6 +39,9 @@ class Actor {
 
   virtual void update_state() = 0;
 
+  virtual void log_shoot(size_t actor_id, size_t opponent_id,
```

Comment:
Can we remove this and maybe make Actor::attack as virtual. And we'll do the logging from there itself appropriately